src/Adapters.Tests/Etw.Net451.Tests/EtwTelemetryModuleTests.cs

```diff
@@ -0,0 +1,218 @@
+                Assert.AreEqual(0, listener.EventsReceived.Count);
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("EtwTelemetryModule")]
+        public void EtwTelemetryModuleProviderEnabledByName()
+        {
+            using (EventSourceModuleDiagnosticListener listener = new EventSourceModuleDiagnosticListener())
+            using (TraceEventSessionMock traceEventSession = new TraceEventSessionMock(true))
+            using (EtwTelemetryModule module = new EtwTelemetryModule(traceEventSession, (t, c) => { }))
+            {
+                module.Sources.Add(new EtwListeningRequest()
+                {
+                    ProviderName = "Test Provider",
+                    Level = Diagnostics.Tracing.TraceEventLevel.Always
+                });
+                module.Initialize(GetTestTelemetryConfiguration());
+                Assert.AreEqual(1, traceEventSession.EnabledProviderNames.Count);
+                Assert.AreEqual("Test Provider", traceEventSession.EnabledProviderNames[0]);
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("EtwTelemetryModule")]
+        public void EtwTelemetryModuleProviderEnabledByGuid()
+        {
+            using (EventSourceModuleDiagnosticListener listener = new EventSourceModuleDiagnosticListener())
+            using (TraceEventSessionMock traceEventSession = new TraceEventSessionMock(true))
+            using (EtwTelemetryModule module = new EtwTelemetryModule(traceEventSession, (t, c) => { }))
+            {
+                Guid guid = Guid.NewGuid();
+                module.Sources.Add(new EtwListeningRequest()
+                {
+                    ProviderGuid = guid,
+                    Level = Diagnostics.Tracing.TraceEventLevel.Always
+                });
+                module.Initialize(GetTestTelemetryConfiguration());
+                Assert.AreEqual(1, traceEventSession.EnabledProviderGuids.Count);
+                Assert.AreEqual(guid.ToString(), traceEventSession.EnabledProviderGuids[0].ToString());
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("EtwTelemetryModule")]
+        public void EtwTelemetryModuleProviderNotEnabledByEmptyGuid()
+        {
+            using (EventSourceModuleDiagnosticListener listener = new EventSourceModuleDiagnosticListener())
+            using (TraceEventSessionMock traceEventSession = new TraceEventSessionMock(true))
+            using (EtwTelemetryModule module = new EtwTelemetryModule(traceEventSession, (t, c) => { }))
+            {
+                Guid guid = Guid.Empty;
+                module.Sources.Add(new EtwListeningRequest()
+                {
+                    ProviderGuid = guid,
+                    Level = Diagnostics.Tracing.TraceEventLevel.Always
+                });
+                module.Initialize(GetTestTelemetryConfiguration());
+                Assert.AreEqual(0, traceEventSession.EnabledProviderGuids.Count);
+            }
+        }
```

Comment:
How about some tests that check

- whether events are actually reported
- whether calling Initialize() with different configuration changes what gets reported
- whether all ETW event properties are correctly translated to TraceTelemetry properties
- whether duplicate property names are handled correctly

TraceEvent is an abstract class so it should be possible to provide your own test implementation and source mock events from test trace session